### ISSUE TITLE
Minor question on comment

### ISSUE THREAD

**Issue by leo-liuzy** [allenai/allennlp#5038]
https://github.com/allenai/allennlp/blob/96415b2bab6d8c70a0fa80ca4a8b9d1dc988720e/allennlp/interpret/saliency_interpreters/integrated_gradient.py#L85

Shouldn't this be a right point approximation? Because we are storing the unmodified output for $\alpha = 0$ (see [here](https://github.com/allenai/allennlp/blob/96415b2bab6d8c70a0fa80ca4a8b9d1dc988720e/allennlp/interpret/saliency_interpreters/integrated_gradient.py#L53)); but this operation is equivalent for $\alpha = 1$.

**Comment by matt-gardner** [comment]
This is breaking the integral into `steps` chunks, and using the top left corner of the rectangle to approximate the integral.  See [here](https://www.emathhelp.net/notes/calculus-2/numerical-approximate-integration/left-endpoint-right-endpoint-and-midpoint-rules/).

**Comment by leo-liuzy** [comment]
I thought for left point approximation, we would have input embedding multiplied by `alpha` taking the value `[0, 0.1, 0.2, ...., 0.9]`. However, what is currently in the code seems to be `[0.1, 0.2, ...., 1]`; this is because of the [line](https://github.com/allenai/allennlp/blob/96415b2bab6d8c70a0fa80ca4a8b9d1dc988720e/allennlp/interpret/saliency_interpreters/integrated_gradient.py#L53) 
```
# Save the input for later use. Only do so on first call.
if alpha == 0:
    embeddings_list.append(output.squeeze(0).clone().detach())
```
To me, the code just replaces `alpha=0` with `alpha=1`. Am I making any misjudgement?

**Comment by matt-gardner** [comment]
If you notice, on that line, we are just saving the unmodified embeddings.  There is no `else` clause.  The multiplication by `alpha` happens below, using the values from the call to `linspace`.  So we get `embeddings * alpha`  (`for alpha in [0, 0.1, ..., 0.9]`) when computing gradients, and we save `embeddings` as-is for later.

**Comment by leo-liuzy** [comment]
Yep, thanks for confirming!!!
